hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockPlacementPolicyDefault.java

```diff
@@ -1236,11 +1245,17 @@ public DatanodeStorageInfo chooseReplicaToDelete(
         minSpace = free;
         minSpaceStorage = storage;
       }
+      if (considerLoad2chooseReplicaDeleting && nodeLoad < minLoad) {
+        minLoad = nodeLoad;
+        minLoadStorage = storage;
+      }
     }
 
     final DatanodeStorageInfo storage;
     if (oldestHeartbeatStorage != null) {
       storage = oldestHeartbeatStorage;
+    } else if (minLoadStorage != null) {
```

Comment:
Good idea to involve load when delete extra replicas. I wonder why we consider load first AND How to trade-offer load vs space? Thanks.

Reply:
@Hexiaoqiao Sir, thanks for your reply.  If we open the configuration mentioned in this PR,  we should consider load first and we keep running Balancer background.  And I also have thinked some improvements about the changes:
1. should we consider load use `load.factor` like `dfs.namenode.redundancy.considerLoad.factor`?
2. should we add another configuration to judge the difference between two storage's remaining in an acceptable range?

Looking forward to your reply~ Thanks sir.